sizefashion/src/context/products_context.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useContext, useEffect, useReducer } from 'react'
+import reducer from '../reducers/products_reducer'
+import {
+  GET_PRODUCTS_BEGIN,
+  GET_PRODUCTS_SUCCESS,
+  GET_PRODUCTS_ERROR,
+  GET_SINGLE_PRODUCT_BEGIN,
+  GET_SINGLE_PRODUCT_SUCCESS,
+  GET_SINGLE_PRODUCT_ERROR,
+} from '../actions'
+import { productDataType } from '../utils/productData'
+import { API_ENDPOINT } from '../utils/constants'
+import axios from 'axios'
+
+export type initialStateType = {
+  allProducts: productDataType[] | []
+  productsLoading: boolean
+  productsError: boolean
+  singleProduct: productDataType | {}
+  singleProductLoading: boolean
+  singleProductError: boolean
+  fetchSingleProduct: (id: string) => void
+}
+
+const initialState: initialStateType = {
+  allProducts: [],
+  productsLoading: false,
+  productsError: false,
+  singleProduct: {},
+  singleProductLoading: false,
+  singleProductError: false,
+  fetchSingleProduct: (id: string) => {},
+}
+
+type Props = {
+  children?: React.ReactNode
+};
+
+export const ProductsContext = React.createContext<initialStateType>(initialState)
+
+export const ProductsProvider: React.FC<Props> = ({ children }) => {
+  
+  const [state, dispatch] = useReducer(reducer, initialState)
+
+  const fetchSingleProduct = (slug: string) => {
+    dispatch({ type: GET_SINGLE_PRODUCT_BEGIN })
+    try {
+      const singleProduct: productDataType = state.allProducts.filter(
+        (product: productDataType) => product.title === slug
+      )[0]
```

Comment:
@otaaviostein using `find` on the array here would work better and require less control over your list.

```js
// Using filter
const whatYouWant = array.filter(item => item.id === yourId)[0]

// Using filter but destructuring the array to return the first one by default
const [whatYouWant] = array.filter(item => item.id === yourId)

// Using find
const whatYouWantV2 = array.find(item => item.id === yourId)
```